### DISCUSSION TITLE
v1.1.0 stable

### DISCUSSION THREAD

**Discussion by tomhea** [tomhea/flip-jump#217] [Announcements]
Version 1.1.0 changes:
- **Total refactor to the standard library** - [cleaner code](https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29), documented better, better tested, and faster in general.
  Thus it has better maintainability and its better for learning.
- All stl macros are now under the stl/bit/hex namespaces
- hex.pointers implemented (faster than bit.ptr)
- Improved READMEs
- Improved debugging (-d, --debuginfo, --debug-ops-list)
- Improve lexer-parser exceptions

It is all tested, and ready to be used on windows or linux (and mac shouldn't be a problem), with python 3.7+.
_The only exception is hex.div which still doesn't work (its tests are xfailed), so you can use the slower-but-stable bit.div+casts for division._

<hr /><em>This discussion was created from the release <a href='https://github.com/tomhea/flip-jump/releases/tag/1.1.0'>v1.1.0 stable</a>.</em>

**Comment by tomhea** [comment]
@lestrozi @vxgmichel @culverit @hmltn2 @jaripekkary @mtve @ebigram @emenel @nst @roma-glushko @Siss3l @tuxcanfly @elhadjx @Peefy @fabiosvm @andrew-johnson-4 @mgmalheiros @frediff @HaraldKorneliussen @adamharmansky @dfischer @CCXXXI @egon-r @Sophia-Cristina @creaktive @Lyxal @parched @TonikHorkel @ivannz @FrogBoy2008 @R-Jones @neauoire @idoherman @danytnt @conilas @gavinbm @jakiki6 @Taardisaa @Clinale @zephryn @thisisclosure @kamil-adam @Polarzz @Sherlockcxk @idozo @betedgi

**Comment by vxgmichel** [comment]
It is indeed faster :rocket: 

This [program](https://github.com/vxgmichel/marbles/blob/main/to-uppercase.fj) (with `a\n` as input) now takes 1364 cycles instead of 1588 before. The program is also smaller, 3062 words instead of 3476 before.

That's a 16% improvement in speed and 14% in size :clap:

**Reply by tomhea** [reply]
Awesome! I did invest lots of time making it better :)
